pangeo_forge_recipes/openers.py

```diff
@@ -45,7 +44,14 @@ def open_url(
 
 def _set_engine(file_type, xr_open_kwargs):
     kw = xr_open_kwargs.copy()
-    if "engine" in kw:
+    if file_type == FileType.unknown:
+        # Enable support for archives containing a mix of e.g. netCDF3 and netCDF4 products etc.
+        warnings.warn(
+            "Unknown file type specified, backend engine will be automatically selected by xarray"
+        )
+        if "engine" in kw:
+            del kw["engine"]
```

Comment:
This is the only part I worry about. Is it really necessary to delete `engine`? I can imagine cases where the user wants to specify an engine and also use `FileType.unknown`. But by default `engine` will already be empty.

Reply:
Thanks @rabernat. I wasn't entirely sure about this either, but I was assuming that based on the following original logic, only engines were accepted where `file_type: engine` was in `OPENERS_MAP`:

https://github.com/pangeo-forge/pangeo-forge-recipes/blob/4b5f1242c9d915ed93b7412c99d6586bbf1cbf70/pangeo_forge_recipes/openers.py#L73-L74

As `unknown` isn't in the map, this would occur (this was the case for me locally before I started modifying it). No problem if we want to retain the `engine` as any issues will be subsequently caught by xarray during opening, I'll commit this now.

Reply:
I think our original assumptions about engine were too strict.

Reply:
I'm not sure that's necessary right now.